### PR TITLE
Disallow variables with built in functions name

### PR DESCRIPTION
**Pull request by ben-kaufman**
Disallow variables to have the name of a built in function.

### - What I did
Didn't have the time to open an issue for that since I want it to be in the beta release.
What do you think?

### - Cute Animal Picture

image

### CODE REVIEW

**Comment by fubuloubu** [comment]
Add a failing testcase (name appears in list).
Bonus points: add a case for a few of them.

LGTM otherwise

File: vyper/utils.py [hunk 1]
```diff
@@ -146,6 +146,16 @@ def in_bounds(cls, type_str, value):
...
 
+built_in_functions = ['floor', 'ceil', 'as_unitless_number',
+                      'convert', 'slice', 'len', 'concat',
+                      'sha3', 'method_id', 'keccak256', 'ecrecover',
+                      'ecadd', 'ecmul', 'extract32', 'as_wei_value',
+                      'raw_call', 'RLPList', 'blockhash', 'bitwise_and',
+                      'bitwise_or', 'bitwise_xor', 'bitwise_not',
+                      'uint256_addmod', 'uint256_mulmod', 'shift',
+                      'create_with_code_of', 'min', 'max', 'send',
+                      'selfdestruct', 'raw_log']
```
Comment (jacqueswww) [inline vyper/utils.py]: We should see if there is not a more dynamic way of pulling these from `functions.py` ?
I believe there is a dict there that could pull this from. Allowing us to not have to worry about changing it in more than one place later.

**Comment by jacqueswww** [comment]
@ben-kaufman thanks for adding this! 

As mentioned above : instead of using a hard coded list we could probably do something like this instead:
```
builtin_functions = [x for x in stmt_dispatch_table.keys()] + [x for x in dispatch_table.keys()]
```

**Comment by ben-kaufman** [comment]
@jacqueswww good idea.
I'm having a weird issue while trying to implement it.
As I do ``import vyper.functions`` the ``make test`` throws this error:
```
ImportError: cannot import name 'get_original_if_0_prefixed'
```
Any chance you know what I did wrong here?

**Comment by jacqueswww** [comment]
Sounds like a recursive import perhaps?
Try importing `from vyper.functions import stmt_dispatch_table` instead of the whole file?

**Comment by ben-kaufman** [comment]
Already tried it... Still the same problem

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by jacqueswww** [comment]
Hmmm that is strange. 🤔

**Comment by jacqueswww** [comment]
How about making a function called `def is_builtin_function(name)` in functions.py and try using that?

**Comment by ben-kaufman** [comment]
This might work, I'll try that later today. If it doesn't work I'll try to push the first way just to make sure that it's not some bug on my machine.

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by jacqueswww** [comment]
I just tested locally, and it is a recursive import. For the sake of completing this PR. Just do this:

```python
# Is a variable or member variable name valid?
def is_varname_valid(varname, custom_units):
    from vyper.functions import dispatch_table, stmt_dispatch_table
    built_in_functions = [x for x in stmt_dispatch_table.keys()] + \
        [x for x in dispatch_table.keys()]
    if custom_units is None:
        custom_units = []
    if varname.lower() in [cu.lower() for cu in custom_units]:
        return False
    if varname.lower() in base_types:
        return False
    if varname.lower() in valid_units:
        return False
    if varname.lower() in reserved_words:
        return False
    if varname.upper() in opcodes:
        return False
    if varname.lower() in built_in_functions:
        return False
    return True
```

**Comment by ben-kaufman** [comment]
Will do that, thanks!

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by ben-kaufman** [comment]
Just fixed it, thanks for the help!

**Comment by jacqueswww** [comment]
LGTM

**Pull request merged**